### PR TITLE
Tequilapi authentication token issuing endpoint

### PR DESCRIPTION
**Pull request by Waldz**
Updates: #2658

This allows Web UI to check is default password is still enabled in node without actually creating auth session.
Also will give more flexibility to Tequilapi's users to create other authentication flows.

### CODE REVIEW

**Review by ignasbernotas — APPROVED** [review]

File: ui/reverse_proxy.go [hunk 1]
```diff
@@ -107,6 +108,43 @@ func ReverseTequilapiProxy(tequilapiAddress string, tequilapiPort int, authentic
...
+	}
+
+	// authenticate from cookie
+	return parseHeaderToken(c)
+}
+
+func parseCookieToken(c *gin.Context) (string, error) {
+	token, err := c.Cookie(auth.JWTCookieName)
+	if err == http.ErrNoCookie {
+		// No error, just no token
```
Comment (tomasmik) [inline ui/reverse_proxy.go:128]: You're repeating the same comment twice here.
Also, if we return no error just an empty string if one occurs, can't we just be a little bit more "adventurous" and ignore the error completely? I mean we're not really handling it and if an errors occurs we'll get an empty string either way.

File: ui/reverse_proxy.go [hunk 2]
```diff
@@ -77,16 +78,16 @@ func ReverseTequilapiProxy(tequilapiAddress string, tequilapiPort int, authentic
 			return
 		}
 
-		// authenticate all but the login route
-		if !isTequilapiURL(c.Request.URL.Path, endpoints.TequilapiLoginEndpointPath) {
-			cookieToken, err := c.Cookie(auth.JWTCookieName)
-
+		// authenticate all but the authentication routes
+		if !(isTequilapiURL(c.Request.URL.Path, endpoints.TequilapiAuthenticateEndpointPath) || isTequilapiURL(c.Request.URL.Path, endpoints.TequilapiLoginEndpointPath)) {
+			// authenticate from header
```
Comment (tomasmik) [inline ui/reverse_proxy.go:83]: This comment is sort of lying as we authenticate from header or cookie

File: ui/reverse_proxy.go [hunk 3]
```diff
@@ -77,16 +78,16 @@ func ReverseTequilapiProxy(tequilapiAddress string, tequilapiPort int, authentic
 			return
 		}
 
-		// authenticate all but the login route
-		if !isTequilapiURL(c.Request.URL.Path, endpoints.TequilapiLoginEndpointPath) {
-			cookieToken, err := c.Cookie(auth.JWTCookieName)
-
+		// authenticate all but the authentication routes
+		if !(isTequilapiURL(c.Request.URL.Path, endpoints.TequilapiAuthenticateEndpointPath) || isTequilapiURL(c.Request.URL.Path, endpoints.TequilapiLoginEndpointPath)) {
```
Comment (tomasmik) [inline ui/reverse_proxy.go:82]: nitpick: could we move this to a self contained func or restructure it some other way? It's kinda hard to read 🤷

**Review by tomasmik — APPROVED** [review]
Disregarding a few nitpicks - looks good.

File: tequilapi/endpoints/auth.go [hunk 4]
```diff
@@ -43,31 +43,70 @@ type authenticator interface {
...
+	}
+	err = api.authenticator.CheckCredentials(req.Username, req.Password)
+	if err != nil {
+		utils.SendError(httpRes, err, http.StatusUnauthorized)
+		return
+	}
+
+	jwtToken, err := api.jwtAuthenticator.CreateToken(req.Username)
+	if err != nil {
+		utils.SendError(httpRes, err, http.StatusBadRequest)
```
[the remark this answers is not in the document]
Comment (Waldz) [reply tequilapi/endpoints/auth.go:80]: you're right, fixed

File: ui/reverse_proxy.go (hunk 1 above)
Comment (Waldz) [reply ui/reverse_proxy.go:128]: Yes, if there is not error i return "".
But with `err` I want to recognise user input error here and return 400 in response

**Review by vkuznecovas — APPROVED** [review]

File: ui/reverse_proxy.go (hunk 3 above)
Comment (Waldz) [reply ui/reverse_proxy.go:82]: Improved

**Pull request merged**